client/tinychain/ml/dnn.py

```diff
@@ -3,21 +3,22 @@
 # Prefer this implementation if no more domain-specific neural net architecture is needed.
 
 from tinychain.collection.tensor import einsum, Dense
-from tinychain.ml import Layer, NeuralNet, Sigmoid
+from tinychain.ml import Layer, NeuralNet, Sigmoid, Parameter
+from tinychain.value import String
 
 
 class DNNLayer(Layer):
     @classmethod
-    def create(cls, input_size, output_size, activation=Sigmoid()):
+    def create(cls, name: str, input_size, output_size, activation=Sigmoid()):
```

Comment:
This breaks the convention where a TinyChain `State` doesn't know it's own name, which is important because TinyChain has no global namespace to check for potential collisions. For example, this would work fine:

```python
def create_layer():
    return DNNLayer("layer", 10, 10)
```

But then this would break:
```python
layers = [create_layer() for _ in range(num_layers)]
```

You could add a "layer_num" parameter to `create_layer` but that still assumes that the calling namespace has only one DNN, etc. If you need a unique identifier at compile time, you can use the `hex_id` function in `tinychain.util`.

Reply:
Actually, I thought about this more. I like the idea of having descriptive names for debugging, we just need to make sure that the top-level calling code (e.g. an Optimizer) can provide a top-level namespace. For example, when an Optimizer is created for a model called "my_nn" it can initialize parameters like "my_nn.0.weights", "my_nn.0.bias" etc. That way if a user uses two instances of my_nn in a larger differentiable module they can be assigned names like "larger_module.my_nn..." and there won't be a collision.

Reply:
The name collision is impossible when the following requirements are fulfilled:
1. The calling code guarantees the unique names of the layers.
2. Every optimizer instance is serving a single model exclusively.
This model/optimizer usage pattern is quite typical for most deep learning frameworks. Are we on the same page or am I missing something?

Reply:
Yes, this sounds fine